homeassistant/components/eq3btsmart/entity.py

```diff
@@ -54,23 +59,13 @@ async def async_added_to_hass(self) -> None:
                 self._async_on_disconnected,
             )
         )
-        self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass,
-                f"{SIGNAL_THERMOSTAT_CONNECTED}_{self._eq3_config.mac_address}",
-                self._async_on_connected,
-            )
-        )
-
-    async def async_will_remove_from_hass(self) -> None:
-        """Run when entity will be removed from hass."""
-
-        self._thermostat.unregister_update_callback(self._async_on_updated)
 
-    def _async_on_updated(self) -> None:
-        """Handle updated data from the thermostat."""
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
 
-        self.async_write_ha_state()
+        self._attr_available = self.coordinator.last_update_success
+        super()._handle_coordinator_update()
```

Comment:
I think you can remove this as well, as this should be in the default implementation of `available` of a `CoordinatorEntity`